File: algorithms/1768.merge-strings-alternately/solution.rs
use std::cmp;

impl Solution {
    pub fn merge_alternately(word1: String, word2: String) -> String {
        let mut res = String::new();
        let len = cmp::max(word1.len(), word2.len());

        let chars1: Vec<char> = word1.chars().collect();
        let chars2: Vec<char> = word2.chars().collect();

        for i in 0..len {
            if i < chars1.len() {
                res.push(chars1[i]);
            }
            if i < chars2.len() {
                res.push(chars2[i]);
            }
        }

        res
    }
}

File: algorithms/1768.merge-strings-alternately/solution.ts
function mergeAlternately(word1: string, word2: string): string {
  let res = "";
  const length = Math.max(word1.length, word2.length);

  for (let i = 0; i < length; i++) {
    if (i < word1.length) {
      res += word1[i];
    }
    if (i < word2.length) {
      res += word2[i];
    }
  }

  return res;
}
